gitlabform/processors/project/branches_processor.py

```diff
@@ -66,28 +67,24 @@ def process_branch_protection(
                 # protected branch is not the same. So, removing existing branch protection and
                 # reconfiguring branch protection seems simpler.
                 self.unprotect_branch(protected_branch)
-                self.protect_branch(project, branch, branch_config)
+                self.protect_branch(project, branch_name, branch_config)
             elif not protected_branch:
-                self.protect_branch(project, branch, branch_config)
+                self.protect_branch(project, branch_name, branch_config)
         elif protected_branch and not branch_config.get("protected"):
             self.unprotect_branch(protected_branch)
 
-    def protect_branch(
-        self, project: Project, branch: ProjectBranch, branch_config: dict
-    ):
+    def protect_branch(self, project: Project, branch: str, branch_config: dict):
```

Comment:
Since you've changed the method signature, (i.e. `branch` is now a string instead of type `ProjectBranch`), this is causing the current test failure for `files` proessor.

https://github.com/gitlabform/gitlabform/actions/runs/11958905461/job/33339482178?pr=887

If this should be a branch name, you'll need to update the `files_process.py` as well and pass the branch name instead of the branch object. I think that should fix test failure.

May I suggest renaming `branch` to `branch_name`? I feel that's a bit clearer what is expected.

Reply:
I believe this may now be sorted, could you take another look?

Reply:
Nice!